servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -60,6 +60,17 @@
     static final int HASH_CODE_SEED = 0xc2b2ae35;
     public static final BiFunction<? super CharSequence, ? super CharSequence, CharSequence> DEFAULT_HEADER_FILTER =
             (k, v) -> "<filtered>";
+    public static final BiFunction<? super CharSequence, ? super CharSequence, CharSequence>
```

Comment:
Make this package private to reduce API surface area. We generally limit creating new public APIs unless they are absolutely required, the intention is to avoid breaking changes (API/behavior) in the future while giving us the flexibility to modify internally used APIs.